applications/callflow/src/module/cf_dynamic_cid.erl

```diff
@@ -104,7 +204,73 @@ handle(Data, Call) ->
 
     {'ok', C1} = cf_exe:get_call(Call),
     Updates = [{fun whapps_call:kvs_store/3, 'dynamic_cid', CID}
-               ,{fun whapps_call:set_caller_id_number/2, CID}
+	      ,{fun whapps_call:set_caller_id_number/2, CID}
               ],
     cf_exe:set_call(whapps_call:exec(Updates, C1)),
     cf_exe:continue(Call).
+
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec should_restrict_call(wh_json:object(), whapps_call:call(), ne_binary()) -> boolean().
+should_restrict_call(Data, Call, Number) ->
+    case wh_json:is_true(<<"enforce_call_restriction">>, Data, 'false') of
+        'false' -> 'false';
+        'true' -> should_restrict_call_by_account(Call, Number)
+    end.
+
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec should_restrict_call_by_account(whapps_call:call(), ne_binary()) -> boolean().
+should_restrict_call_by_account(Call, Number) ->
+    AccountId = whapps_call:account_id(Call),
+    AccountDb = whapps_call:account_db(Call),
+    case couch_mgr:open_cache_doc(AccountDb, AccountId) of
```

Comment:
@stormqloud 
Why just by account?  Are you sure that you do not want to check the authorization based on the endpoint again?  For example, we allow the device or user to be restricted so if it is set at this level this module could  bypass that restriction if only the account is checked....

If this was not intentionally I would recommend using:
cf_endpoint:get(Call)
this will give you a JSON object that is the result of some merge logic so that it represents the settings for the device, user and account.  Getting the call restriction from this object will be identical to the restrictions enforced when calls are initially set up:
Here: https://github.com/2600hz/kazoo/blob/master/applications/callflow/src/cf_route_win.erl#L55
the merged object is fetched (a 'endpoint' object is the merge of settings from the dialing device, user if associated, and account).  This is then passed through a series of functions until it gets here: https://github.com/2600hz/kazoo/blob/master/applications/callflow/src/cf_route_win.erl#L100 as the JObj property.